### PR TITLE
tinygo: fix reproducible vendor

### PR DESCRIPTION
**Pull request by kirillrdy**
###### Description of changes

this is in relation to https://github.com/NixOS/nixpkgs/pull/245159

it looks like modifying (writing to ) `$out` in `postPatch` causes vendor hash to change

I've only tested using building wasm binaries

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by kirillrdy** [comment]
investigating build failure on darwin

File: pkgs/development/compilers/tinygo/default.nix [hunk 1]
```diff
@@ -101,12 +101,7 @@ buildGoModule rec {
     sed -i "/GOOS=windows/d" Makefile
     sed -i "/GOOS=darwin/d" Makefile
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/tinygo/default.nix:103]:
```suggestion
```

File: pkgs/development/compilers/tinygo/default.nix [hunk 2]
```diff
@@ -101,12 +101,7 @@ buildGoModule rec {
...
 
-    # tinygo needs versioned binaries
-    mkdir -p $out/libexec/tinygo
-    ln -s ${lib.getBin clang.cc}/bin/clang $out/libexec/tinygo/clang-${llvmMajor}
-    ln -s ${lib.getBin lld}/bin/ld.lld $out/libexec/tinygo/ld.lld-${llvmMajor}
-    ln -s ${lib.getBin lld}/bin/wasm-ld $out/libexec/tinygo/wasm-ld-${llvmMajor}
-    ln -s ${gdb}/bin/gdb $out/libexec/tinygo/gdb-multiarch
+
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/tinygo/default.nix:104]:
```suggestion
```

Comment (kirillrdy) [reply pkgs/development/compilers/tinygo/default.nix:104]: @SuperSandro2000 will do
can you fix Darwin build :smile:

Comment (SuperSandro2000) [reply pkgs/development/compilers/tinygo/default.nix:104]: Try including the package including the `stddef.h` header file. Not sure which one that is, you can try finding it via nix-locate.

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by kirillrdy** [comment]
@Janik-Haag can you have a look a this?

**Comment by Janik-Haag** [comment]
Result of `nixpkgs-review pr 245445` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tinygo</li>
  </ul>
</details>

**Review by Janik-Haag — APPROVED** [review]
LGTM

**Pull request merged**